Спринт 9

cyberschoolinmoscow opened the pull request:
1) Созданы общие стили для атрибутов на «Главном экране» и в «Настройках».
2) Добавлены новые шрифты в экраны.
3) Адаптирован интерфейс под «Ночной режим».
4) При включении ночного режима приложение меняет тему на тёмную.
5) Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».
6) Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.
7) Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

cyberschoolinmoscow committed:
Изменена тема с Material3 на MaterialComponents

cyberschoolinmoscow committed:
Изменена тема с Material3 на MaterialComponents для values-night

cyberschoolinmoscow committed:
Добавлены layout\activity_main, иконки для кнопок

cyberschoolinmoscow committed:
Добавлены layout\activity_settings и values\styles

cyberschoolinmoscow committed:
Добавлены пустые MediatekaActivity и SearchActivity, переходы из MainActivity в MediatekaActivity, SearchActivity, SettingsActivity, переход из SettingsActivity в MainActivity.

cyberschoolinmoscow committed:
1) Созданы общие стили для атрибутов на «Главном экране» и в «Настройках».
2) Добавлены новые шрифты в экраны.
3) Адаптирован интерфейс под «Ночной режим».
4) При включении ночного режима приложение меняет тему на тёмную.
5) Реализованы кнопки «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение».
6) Реализован экран «Поиск» без выполнения поискового запроса и отображения списка результатов.
7) Реализовано сохранение текста поискового запроса в жизненном цикле SearchActivity.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,70 @@
+import androidx.appcompat.app.AppCompatActivity
+
+class SearchActivity : AppCompatActivity() {
+    lateinit var inputEditText: EditText
```
private

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,70 @@
+
+class SearchActivity : AppCompatActivity() {
+    lateinit var inputEditText: EditText
+    val INPUT_STRING = "input"
```
Строки-ключи можно выносить в константы. В данном случае в companion object класса.

Это не то чтобы обязательно, но так принято.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,70 @@
+
+        inputEditText = findViewById(R.id.et_search)
+        val clearButton: ImageView = findViewById(R.id.iv_clear)
+        clearButton.visibility = ImageView.GONE
```
изначальную видимость лучше задать в xml-файле лейаута

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,70 @@
+        }
+        inputEditText.addTextChangedListener(simpleTextWatcher)
+
+        var myToolbar = findViewById<androidx.appcompat.widget.Toolbar>(R.id.idSearchToolbar)
```
val

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,70 @@
+        return if (s.isNullOrEmpty()) {
+            View.GONE
+        } else {
+            View.VISIBLE
```
Для изменения видимости элемента на экране в данном случае удобнее использовать extension View.isVisible:
View.isVisible = true соответствует View.visibility = View.VISIBLE
View.isVisible = false соответствует View.visibility = View.GONE

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,15 @@
+    <dimen name="margin_vertical">16dp</dimen>
+    <dimen name="corner_radius">16dp</dimen>
+    <dimen name="layout_marginTop">20dp</dimen>
+    <dimen name="layout_height_settings_button">61dp</dimen>
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). По крайней мере чётными. Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,106 @@
+        <item name="tint">?attr/colorBackgroundFloating</item>
+    </style>
+
+
```
Много лишнего в проекте вынесено в стили. Лучше внешние отступы, высоту и ширину, веса, гравити не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи и вообще только за то, что находится внутри нее - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

cyberschoolinmoscow replied:
Спасибо

cyberschoolinmoscow committed:
1) Созданы классы, описывающие элемент списка, Track, TrackViewHolder и TrackAdapter.
2) Для загрузки обложки трека используется Glide.
3) Элементы списка сверстаны.
